app/server/lib/ExportXLSX.ts

```diff
@@ -1,34 +1,108 @@
 import {ActiveDoc} from 'app/server/lib/ActiveDoc';
 import {createExcelFormatter} from 'app/server/lib/ExcelFormatter';
-import {ExportData, exportDoc} from 'app/server/lib/Export';
+import {ExportData, exportDoc, ExportParameters, exportSection, exportTable, Filter} from 'app/server/lib/Export';
 import {Alignment, Border, Fill, Workbook} from 'exceljs';
 import * as express from 'express';
 import log from 'app/server/lib/log';
 import contentDisposition from 'content-disposition';
+import { ApiError } from 'app/common/ApiError';
 
-export interface DownloadXLSXOptions {
+export interface DownloadXLSXOptions extends ExportParameters {
   filename: string;
 }
 
+export function buildDownloadXlsxOptions({
+  filename,
+  tableId,
+  viewSectionId,
+  filters,
+  sortOrder,
+  activeView,
+}: any): DownloadXLSXOptions {
+  return {
+    filename,
+    tableId: tableId || '',
+    viewSectionId: viewSectionId || undefined,
+    filters: filters || [],
+    sortOrder: sortOrder || [],
+    activeView: activeView || false,
+  }
+}
+
 /**
- * Converts `activeDoc` to CSV and sends the converted data through `res`.
+ * Converts `activeDoc` to XLSX and sends the converted data through `res`.
  */
 export async function downloadXLSX(activeDoc: ActiveDoc, req: express.Request,
-                                   res: express.Response, {filename}: DownloadXLSXOptions) {
+                                   res: express.Response, options: DownloadXLSXOptions) {
   log.debug(`Generating .xlsx file`);
-  const data = await makeXLSX(activeDoc, req);
+  const {filename, tableId, viewSectionId, filters, sortOrder, activeView} = options;
+  // hanlding 3 cases : full XLSX export (full file), view xlsx export, table xlsx export
+  const data = !activeView ? await makeXLSX(activeDoc, req)
+              : viewSectionId ? await makeXLSXFromViewSection(activeDoc, viewSectionId, sortOrder, filters, req)
+              : await makeXLSXFromTable(activeDoc, tableId, req);
+
   res.set('Content-Type', 'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet');
   res.setHeader('Content-Disposition', contentDisposition(filename + '.xlsx'));
   res.send(data);
   log.debug('XLSX file generated');
 }
 
+/**
+ * Returns a XLSX stream of a view section that can be transformed or parsed.
+ *
+ * @param {Object} activeDoc - the activeDoc that the table being converted belongs to.
+ * @param {Integer} viewSectionId - id of the viewsection to export.
+ * @param {Integer[]} activeSortOrder (optional) - overriding sort order.
+ * @param {Filter[]} filters (optional) - filters defined from ui.
+ */
+ export async function makeXLSXFromViewSection(
+  activeDoc: ActiveDoc,
+  viewSectionId: number,
+  sortOrder: number[],
+  filters: Filter[],
+  req: express.Request,
+) {
+
+  const data = await exportSection(activeDoc, viewSectionId, sortOrder, filters, req);
+  const xlsx = await convertToExcel([data], req.hostname === 'localhost');
```

Comment:
@berhalak the `localhost` handling in `ExportXLSX` is there for tests, is that right? Maybe we should file an issue to clean it up via a dependency injection or something, what do you think? It doesn't feel great to see these `localhost` conditionals spreading out.

Reply:
Do you want me to handle this or should I let you handle it ?
I'd prefer to let you find a way to manage this testing variable, if you didn't already have one.

Reply:
(Yes @LouisDelbosc you can ignore this issue, we'll take care of it if there is something that needs doing)